gui/src/application_restarter.cpp

```diff
@@ -0,0 +1,31 @@
+#include "application_restarter.hpp"
+
+#include <QApplication>
+#include <QDir>
+#include <QProcess>
+
+using namespace scopy::gui;
+
+ApplicationRestarter::ApplicationRestarter(const QString& executable)
```

Comment:
I think that the application restarter should be in scopy::core as its not a GUI related class